core/src/smartcontracts/wasm.rs

```diff
@@ -526,18 +628,133 @@ mod tests {
             isi1_end = isi_hex.len() / 3,
             isi2_end = 2 * isi_hex.len() / 3,
         );
-        let mut runtime = Runtime::from_configuration(Configuration {
-            fuel_limit: 100_000,
-            max_instruction_count: 1,
-        })?;
-        let res = runtime.execute(&wsv, account_id, wat);
+
+        let mut runtime = Runtime::new()?;
+        let res = runtime.validate(
+            &wsv,
+            &account_id,
+            wat,
+            1,
+            Arc::new(AllowAll.into()),
+            Arc::new(AllowAll.into()),
+        );
+
+        assert!(res.is_err());
+        if let Error::ExportFnCall(trap) = res.unwrap_err() {
+            assert!(trap
+                .display_reason()
+                .to_string()
+                .starts_with("Number of instructions exceeds maximum(1)"));
```

Comment:
In the future this should be `FromStr` for the error variant. Relying on a specific non-constant string is a bit problematic.